Analyses/Simulation/commands/sim_part_1_settings.txt
Control File Settings:
tree_length        [ 5.0, 5.0, 15.0, 5.0, 5.0, 15.0 ]
base_frequencies   [ (0.1, 0.4, 0.4, 0.1), (0.25, 0.25, 0.25, 0.25), (0.4, 0.1, 0.1, 0.4), (0.1, 0.4, 0.4, 0.1), (0.25, 0.25, 0.25, 0.25), (0.4, 0.1, 0.1, 0.4) ]
substitution_rates [ (1.0, 2.0, 1.0, 1.0, 2.0, 1.0), (1.0, 2.0, 1.0, 1.0, 2.0, 1.0), (1.0, 2.0, 1.0, 1.0, 2.0, 1.0), (2.0, 1.0, 1.0, 1.0, 1.0, 2.0), (2.0, 1.0, 1.0, 1.0, 1.0, 2.0), (2.0, 1.0, 1.0, 1.0, 1.0, 2.0) ]
gamma_shape        [ 0.1, 0.01, 0.9, 0.9, 0.01, 0.1 ]
number_sites       [ 1000, 1000, 1000, 1000, 1000, 1000 ]
num_taxa           [ 50 ]
num_reps           [ 100 ]


AutoPartsSim Screen log:
Number of replicates: 100
Number of taxa:       50
Branch length prior:  5 5 15 5 5 15 
Gamma shape:          0.1 0.01 0.9 0.9 0.01 0.1 
Number of sites:      1000 1000 1000 1000 1000 1000 
Number of sites:      ( 0.1 0.4 0.4 0.1 ) ( 0.25 0.25 0.25 0.25 ) ( 0.4 0.1 0.1 0.4 ) ( 0.1 0.4 0.4 0.1 ) ( 0.25 0.25 0.25 0.25 ) ( 0.4 0.1 0.1 0.4 ) 
Substitution rates:   ( 1 2 1 1 2 1 ) ( 1 2 1 1 2 1 ) ( 1 2 1 1 2 1 ) ( 2 1 1 1 1 2 ) ( 2 1 1 1 1 2 ) ( 2 1 1 1 1 2 ) 
Tree: 
0 -- 0 (51 -1 -1) 0.0000 <- Root
1 -- 1 (-1 -1 50) 0.0041
2 -- 2 (-1 -1 52) 0.0035
3 -- 3 (-1 -1 97) 0.0000
4 -- 4 (-1 -1 74) 0.0054
5 -- 5 (-1 -1 57) 0.0041
6 -- 6 (-1 -1 64) 0.0044
7 -- 7 (-1 -1 55) 0.0039
8 -- 8 (-1 -1 93) 0.0032
9 -- 9 (-1 -1 68) 0.0062
10 -- 10 (-1 -1 89) 0.0256
11 -- 11 (-1 -1 70) 0.0015
12 -- 12 (-1 -1 62) 0.0117
13 -- 13 (-1 -1 61) 0.0054
14 -- 14 (-1 -1 62) 0.0073
15 -- 15 (-1 -1 78) 0.0082
16 -- 16 (-1 -1 81) 0.0016
17 -- 17 (-1 -1 65) 0.0113
18 -- 18 (-1 -1 66) 0.0154
19 -- 19 (-1 -1 67) 0.0162
20 -- 20 (-1 -1 71) 0.0327
21 -- 21 (-1 -1 84) 0.0188
22 -- 22 (-1 -1 70) 0.0228
23 -- 23 (-1 -1 82) 0.0185
24 -- 24 (-1 -1 72) 0.0029
25 -- 25 (-1 -1 73) 0.0113
26 -- 26 (-1 -1 74) 0.0064
27 -- 27 (-1 -1 75) 0.0120
28 -- 28 (-1 -1 76) 0.0110
29 -- 29 (-1 -1 83) 0.0064
30 -- 30 (-1 -1 78) 0.0408
31 -- 31 (-1 -1 79) 0.0206
32 -- 32 (-1 -1 80) 0.0007
33 -- 33 (-1 -1 85) 0.0154
34 -- 34 (-1 -1 82) 0.0021
35 -- 35 (-1 -1 87) 0.0024
36 -- 36 (-1 -1 84) 0.0329
37 -- 37 (-1 -1 85) 0.0013
38 -- 38 (-1 -1 86) 0.0008
39 -- 39 (-1 -1 87) 0.0048
40 -- 40 (-1 -1 88) 0.0054
41 -- 41 (-1 -1 95) 0.0063
42 -- 42 (-1 -1 96) 0.0228
43 -- 43 (-1 -1 91) 0.0028
44 -- 44 (-1 -1 92) 0.0154
45 -- 45 (-1 -1 93) 0.0263
46 -- 46 (-1 -1 94) 0.0015
47 -- 47 (-1 -1 95) 0.0024
48 -- 48 (-1 -1 96) 0.0165
49 -- 49 (-1 -1 97) 0.0042
50 -- 50 (1 61 77) 0.0062
51 -- 51 (54 97 0) 0.0317
52 -- 52 (66 2 79) 0.0008
53 -- 53 (67 79 94) 0.0002
54 -- 54 (63 65 51) 0.0184
55 -- 55 (7 64 73) 0.0058
56 -- 56 (90 93 58) 0.0000
57 -- 57 (5 68 67) 0.0009
58 -- 58 (56 89 66) 0.0184
59 -- 59 (60 73 92) 0.0156
60 -- 60 (72 76 59) 0.0035
61 -- 61 (13 94 50) 0.0073
62 -- 62 (14 12 76) 0.0084
63 -- 63 (77 78 54) 0.0022
64 -- 64 (81 6 55) 0.0006
65 -- 65 (17 80 54) 0.0133
66 -- 66 (58 18 52) 0.0216
67 -- 67 (57 19 53) 0.0026
68 -- 68 (69 9 57) 0.0005
69 -- 69 (91 88 68) 0.0009
70 -- 70 (11 22 72) 0.0013
71 -- 71 (20 75 91) 0.0045
72 -- 72 (70 24 60) 0.0317
73 -- 73 (25 55 59) 0.0110
74 -- 74 (4 26 90) 0.0150
75 -- 75 (27 86 71) 0.0146
76 -- 76 (28 62 60) 0.0010
77 -- 77 (50 83 63) 0.0091
78 -- 78 (30 15 63) 0.0200
79 -- 79 (31 52 53) 0.0155
80 -- 80 (32 92 65) 0.0122
81 -- 81 (16 85 64) 0.0004
82 -- 82 (34 23 86) 0.0212
83 -- 83 (87 29 77) 0.0330
84 -- 84 (36 21 88) 0.0183
85 -- 85 (37 33 81) 0.0242
86 -- 86 (38 82 75) 0.0030
87 -- 87 (35 39 83) 0.0030
88 -- 88 (40 84 69) 0.0103
89 -- 89 (95 10 58) 0.0235
90 -- 90 (74 96 56) 0.0048
91 -- 91 (71 43 69) 0.0150
92 -- 92 (44 59 80) 0.0144
93 -- 93 (45 8 56) 0.0023
94 -- 94 (46 53 61) 0.0077
95 -- 95 (41 47 89) 0.0037
96 -- 96 (48 42 90) 0.0101
97 -- 97 (49 3 51) 0.0029


Analyses/Simulation/output/source/Settings.cpp
#include <iostream>
#include <cmath>
#include <string>
#include "Settings.h"



Settings::Settings(int argc, char *argv[]) {

#	if 0
	/* set up fake command-line argument string */
	char *cmdString[27];
	cmdString[ 0] = (char*)"ap";
	cmdString[ 1] = (char*)"-i";
	cmdString[ 2] = (char*)"/Users/brianmoore/Documents/Briofile/AutoParts/AutoParts/source/test.in";
	cmdString[ 3] = (char*)"-o";
	cmdString[ 4] = (char*)"/Users/brianmoore/Documents/Briofile/AutoParts/AutoParts/source/test_smart";
	cmdString[ 5] = (char*)"-l";
	cmdString[ 6] = (char*)"100";
	cmdString[ 7] = (char*)"-p";
	cmdString[ 8] = (char*)"10";
	cmdString[ 9] = (char*)"-s";
	cmdString[10] = (char*)"10";
	cmdString[11] = (char*)"-b";
	cmdString[12] = (char*)"20.0";
	cmdString[13] = (char*)"-g";
	cmdString[14] = (char*)"4";
	cmdString[15] = (char*)"-e";
	cmdString[16] = (char*)"2.0";
	cmdString[17] = (char*)"-c";
	cmdString[18] = (char*)"1";
	cmdString[19] = (char*)"-m";
	cmdString[20] = (char*)"1.1";
	cmdString[21] = (char*)"-v";
	cmdString[22] = (char*)"2.0";
	cmdString[23] = (char*)"-k";
	cmdString[24] = (char*)"3.0";
	cmdString[25] = (char*)"-d";
	cmdString[26] = (char*)"50000";
	//cmdString[23] = "-t";
	//cmdString[24] = "/Users/johnh/Working/chungking/.tree";
	argc = 27;
	argv = cmdString;
#	endif

	enum Mode { DATA_FILE, TREE_FILE, OUTPUT_FILE, BURN_IN, CHAIN_LENGTH, PRINT_FREQ, SAMPLE_FREQ, BRLEN_PARM, NUM_GAMMA_CATS, ASRV_LAMBDA, CONC_FIXED, EXP_CATS, CONC_MEAN, CONC_VAR, TUNE_T1, TUNE_T2, TUNE_T3, TUNE_T4, TUNE_T5, NONE };

	/* set default values for parameters */
	dataFilePathName       = "";
	treeFileName           = "";
	outPutFileName         = "";
	chainLength            = 1000000;
	burnIn                 = 0;
	printFrequency         = 100;
	sampleFrequency        = 100;
	brlenLambda            = 10.0;
	numGammaCats           = 4;
	asrvLambda             = 2.0;
	isConcFixed            = true;
	expNumCats             = 3.0;
	priorConcMean          = 3.0;
	priorConcVariance      = 1.0;
	tuningParm[0]          = 500.0;
	tuningParm[1]          = log(1.4);
	tuningParm[2]          = 300.0;
	tuningParm[3]          = 300.0;
	tuningParm[4]          = log(1.2);
	
	if (argc > 1)
		{
		if (argc % 2 == 0)
			{
			printUsage();
			}
			
		/* read the command-line arguments */
		int status = NONE;
		for (int i=1; i<argc; i++)
			{
			std::string cmd = argv[i];
			//std::cout << cmd << std::endl;
			if (status == NONE)
				{
				/* read the parameter specifier */
				if ( cmd == "-i" )
					status = DATA_FILE;
				else if ( cmd == "-t" )
					status = TREE_FILE;
				else if ( cmd == "-o" )
					status = OUTPUT_FILE;
				else if ( cmd == "-l" )
					status = CHAIN_LENGTH;
				else if ( cmd == "-d" )
					status = BURN_IN;
				else if ( cmd == "-p" )
					status = PRINT_FREQ;
				else if ( cmd == "-s" )
					status = SAMPLE_FREQ;
				else if ( cmd == "-b" )
					status = BRLEN_PARM;
				else if ( cmd == "-g" )
					status = NUM_GAMMA_CATS;
				else if ( cmd == "-e" )
					status = ASRV_LAMBDA;
				else if ( cmd == "-c" )
					status = CONC_FIXED;
				else if ( cmd == "-k" )
					status = EXP_CATS;
				else if ( cmd == "-m" )
					status = CONC_MEAN;
				else if ( cmd == "-v" )
					status = CONC_VAR;
				else if ( cmd == "-t1" )
					status = TUNE_T1;
				else if ( cmd == "-t2" )
					status = TUNE_T2;
				else if ( cmd == "-t3" )
					status = TUNE_T3;
				else if ( cmd == "-t4" )
					status = TUNE_T4;
				else if ( cmd == "-t5" )
					status = TUNE_T5;
				else
					{
					std::cerr << "Could not interpret option \"" << cmd << "\"." << std::endl;
					exit(1);
					}
				}
			else
				{
				/* read the parameter */
				if ( status == DATA_FILE )
					dataFilePathName = argv[i];
				else if ( status == OUTPUT_FILE )
					outPutFileName = argv[i];
				else if ( status == TREE_FILE )
					treeFileName = argv[i];
				else if ( status == CHAIN_LENGTH )
					chainLength = atoi(argv[i]);
				else if ( status == BURN_IN )
					burnIn = atoi(argv[i]);
				else if ( status == PRINT_FREQ )
					printFrequency = atoi(argv[i]);
				else if ( status == SAMPLE_FREQ )
					sampleFrequency = atoi(argv[i]);
				else if ( status == BRLEN_PARM )
					brlenLambda = atof(argv[i]);
				else if ( status == NUM_GAMMA_CATS )
					numGammaCats = atoi(argv[i]);
				else if ( status == ASRV_LAMBDA )
					asrvLambda = atof(argv[i]);
				else if ( status == CONC_FIXED )
					isConcFixed = atoi(argv[i]);
				else if ( status == EXP_CATS )
					expNumCats = atof(argv[i]);
				else if ( status == CONC_MEAN )
					priorConcMean = atof(argv[i]);
				else if ( status == CONC_VAR )
					priorConcVariance = atof(argv[i]);
				else if ( status == TUNE_T1 )
					tuningParm[0] = atof(argv[i]);
				else if ( status == TUNE_T2 )
					tuningParm[1] = atof(argv[i]);
				else if ( status == TUNE_T3 )
					tuningParm[2] = atof(argv[i]);
				else if ( status == TUNE_T4 )
					tuningParm[3] = atof(argv[i]);
				else if ( status == TUNE_T5 )
					tuningParm[4] = atof(argv[i]);
				else
					{
					std::cerr << "Unknown status reading command line information" << std::endl;
					exit(1);
					}
				status = NONE;
				}
			}
		}
	else
		{
		printUsage();
		}	

}

double Settings::getTuningParm(std::string parmNameStr) {

	if ( parmNameStr == "tree" )
		return tuningParm[0];
	else if ( parmNameStr == "asrv" )
		return tuningParm[1];
	else if ( parmNameStr == "freqs" )
		return tuningParm[2];
	else if ( parmNameStr == "rates" )
		return tuningParm[3];
	else if ( parmNameStr == "length" )
		return tuningParm[4];
	else
		{
		std::cerr << "ERROR: Unknown parameter " << parmNameStr << std::endl;
		exit(1);
		}
	return 0.0;
}

void Settings::printUsage(void) {

	std::cout << "Usage:" << std::endl;
	std::cout << "   -i : Input file name" << std::endl;
	std::cout << "   -t : Tree file name (for constraining the analysis to a fixed tree)" << std::endl;
	std::cout << "   -o : Output file name" << std::endl;
	std::cout << "   -l : Number of MCMC cycles" << std::endl;
	std::cout << "   -d : Number of MCMC cycles to discard as the \"burn-in\"" << std::endl;
	std::cout << "   -p : Print frequency" << std::endl;
	std::cout << "   -s : Sample frequency" << std::endl;
	std::cout << "   -b : Exponential parameter for branch lengths" << std::endl;
	std::cout << "   -g : Number of discrete gamma rate categories" << std::endl;
	std::cout << "   -e : Exponential parameter for shape parameter describing ASRV" << std::endl;
	std::cout << "   -c : Concentration parameter is fixed (1) or a random variable (0)" << std::endl;
	std::cout << "   -k : Prior mean of the number of categories when the concentration parameter is fixed" << std::endl;
	std::cout << "   -m : Prior mean of the concentration parameter when it is a random variable" << std::endl;
	std::cout << "   -v : Prior variance of the concentration parameter when it is a random variable" << std::endl;
	std::cout << "  -t1 : MCMC tuning parameter for the tree topology parameter" << std::endl;
	std::cout << "  -t2 : MCMC tuning parameter for the gamma shape parameter" << std::endl;
	std::cout << "  -t3 : MCMC tuning parameter for the base frequencies" << std::endl;
	std::cout << "  -t4 : MCMC tuning parameter for the substitution rates" << std::endl;
	std::cout << "  -t5 : MCMC tuning parameter for the tree length parameter" << std::endl;
	std::cout << std::endl;
	std::cout << "Example:" << std::endl;
	std::cout << "   ./AutoParts -i <input file> -o <output file>" << std::endl;
	exit(0);

}